silkworm/silkrpc/core/evm_trace.cpp

```diff
@@ -1380,6 +1393,28 @@ awaitable<std::vector<Trace>> TraceCallExecutor::trace_transaction(const BlockWi
     co_return traces;
 }
 
+boost::asio::awaitable<TraceEntriesResult> TraceCallExecutor::trace_ots_transaction(const TransactionWithBlock& transaction_with_block) {
+    auto block_number = transaction_with_block.block_with_hash.block.header.number;
+
+    const auto chain_id = co_await core::rawdb::read_chain_id(database_reader_);
+    const auto chain_config_ptr = lookup_chain_config(chain_id);
+
+    state::RemoteState remote_state{io_context_, database_reader_, block_number - 1};
+    silkworm::IntraBlockState initial_ibs{remote_state};
+
+    state::RemoteState curr_remote_state{io_context_, database_reader_, block_number - 1};
+    EVMExecutor executor{*chain_config_ptr, workers_, curr_remote_state};
+
+    auto entry_tracer = std::make_shared<trace::EntryTracer>(initial_ibs);
+
+    Tracers tracers{entry_tracer};
+
+    co_await executor.call(transaction_with_block.block_with_hash.block, transaction_with_block.transaction, tracers, /*refund=*/true, /*gas_bailout=*/true);
+    executor.reset();
```

Comment:
Probably no need to reset the `executor` because it isn't used anymore and will be destroyed on function exit